packages/alea-frontend/pages/api/university-admin/semester-info/create-semester.ts

```diff
@@ -0,0 +1,70 @@
+import { Action, ResourceName } from '@stex-react/utils';
+import { getUserIdIfAuthorizedOrSetError } from '../../access-control/resource-utils';
+import { executeQuery, checkIfPostOrSetError } from '../../comment-utils';
+
+export default async function handler(req, res) {
+  if (!checkIfPostOrSetError(req, res)) return;
+
+  const userId = await getUserIdIfAuthorizedOrSetError(req,res,ResourceName.UNIVERSITY_SEMESTER_DATA,Action.MUTATE,{universityId: req.query.universityId});
+  if (!userId) return;
+
+  const {
+    universityId,
+    instanceId,
+    semesterStart,
+    semesterEnd,
+    lectureStartDate,
+    lectureEndDate,
+    timeZone,
+  } = req.body;
+
+  if (
+    !universityId ||
+    !instanceId ||
+    !semesterStart ||
+    !semesterEnd ||
+    !lectureStartDate ||
+    !lectureEndDate ||
+    !timeZone
+  ) {
+    return res.status(400).json({ message: 'Missing required fields' });
+  }
+
+  const existing = (await executeQuery(
+    `SELECT 1 FROM semesterInfo WHERE universityId = ? AND instanceId = ?`,
+    [universityId, instanceId]
+  )) as any[] | { error: any };
+
+  if ('error' in existing) {
+    return res.status(500).json({ message: 'Database error', error: existing.error });
+  }
+

```

Comment:
Please read the guidelines ("Writing APIs") very carefully https://github.com/KWARC/ALeA/blob/main/backend_and_general_guidelines.md#2-writing-apis

In this case, you can very conveniently, use `executeDontEndSet500OnError` 

```suggestion
  const existing = (await executeDontEndSet500OnError(
    `SELECT 1 FROM semesterInfo WHERE universityId = ? AND instanceId = ?`,
    [universityId, instanceId]
  )) as any[] | { error: any };

  if (!existing) return;

```